credentials/tls.go

```diff
@@ -25,14 +25,16 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net"
+	"net/url"
 
 	"google.golang.org/grpc/credentials/internal"
 )
 
 // TLSInfo contains the auth information for a TLS authenticated connection.
 // It implements the AuthInfo interface.
 type TLSInfo struct {
-	State tls.ConnectionState
+	State    tls.ConnectionState
+	SpiffeID *url.URL
```

Comment:
Have we considered String? 
Ultimately, we just need it as string for authorization check.

Reply:
Discussed offline: we will keep the SpiffeID as the `*url.URL` field and users could convert the field to `string` after they get this information.